common/console.cpp

```diff
@@ -52,6 +53,21 @@ namespace console {
     static termios   initial_state;
 #endif
 
+#if !defined(_WIN32)
+    static bool locale_isvalid( const char * l )
+    {
+        return !(l == nullptr || l[0] == '\0' || (strcasestr(l, "utf-8") == nullptr && strcasestr(l, "utf8") == nullptr));
+    }
+
+    static bool locale_setverify( int c, const char * l )
+    {
+        setlocale(c, l);
+        auto locale = setlocale(c, nullptr);
+
+        return !strcmp( l, locale );
+    }
+#endif
```

Comment:
> Turns out setlocale(X,"") is not the same as setlocale(X,nullptr) as the former only returns nullptr if locale changes, whereas the latter returns but doesn't change.

You are right about passing in NULL for the second argument, however, you are wrong about how it works when you pass in a string. `setlocale` will only return `nullptr` if it fails to set the locale. It will also *always* return the current locale if it succeeds, and it will succeed even if you pass it in the current locale and nothing changes. So there is no need for a separate call to check the results. You can simply look at the return results.

`locale_setverify` is unnecessary at best and wrong at worst. It won't work correctly when you pass in `l` of empty string (`""`) for example - which `getenv("LANG")` could absolutely be.